eox_core/api/data/data_collector/utils.py

```diff
@@ -0,0 +1,109 @@
+"""
+Utility functions for report generation, including query execution
+and integration with the Shipyard API.
+"""
+
+import yaml
+from django.db import connection
+import requests
+from django.conf import settings
+from datetime import datetime
+import logging
+import re
+
+from eox_core.utils import get_access_token
+
+logger = logging.getLogger(__name__)
+
+
+def execute_query(sql_query):
+    """
+    Execute a raw SQL query and return the results in a structured format.
+
+    Args:
+        sql_query (str): The raw SQL query to execute.
+
+    Returns:
+        list or dict: Structured query results.
+
+    Raises:
+        ValueError: If the query is not a SELECT statement.
+    """
+    # Normalize query (remove whitespace and convert to uppercase)
+    normalized_query = sql_query.strip().upper()
+
+    # Verify that the query begins with "SELECT"
+    if not re.match(r"^SELECT\s", normalized_query):
+        raise ValueError("Only SELECT queries are allowed.")
+
+    with connection.cursor() as cursor:
+        cursor.execute(sql_query)
+        rows = cursor.fetchall()
+        # If the query returns more than one column, return rows as is.
+        if cursor.description:
+            columns = [col[0] for col in cursor.description]
+            if len(columns) == 1:
+                return [row[0] for row in rows]  # Return single-column results as a list
+            return [dict(zip(columns, row)) for row in rows]  # Multi-column results as a list of dicts
+        return rows
+
+
+def serialize_data(data):
```

Comment:
I think these two functions `serialize_data` and `process_query_results` could get merged into a single function called `clean_query_results()` or `post_process_query_results()`